src/implementationGuides/IGCompiler.ts

```diff
@@ -83,6 +90,23 @@ export class IGCompiler {
         return resources;
     }
 
+    async collectStructureDefinitions(igDir: PathLike): Promise<any[]> {
+        const indexJson = path.join(igDir.toString(), '.index.json');
+        if (!existsSync(indexJson)) {
+            throw new Error(`'.index.json' not found in ${igDir}`);
+        }
+        const index: any = await loadJson(indexJson);
+        const resources = [];
+        for (const file of index.files) {
+            if (file.resourceType === 'StructureDefinition') {
+                const filePath = path.join(igDir.toString(), file.filename);
+                console.log(`Compiling ${filePath}`);
+                resources.push(await loadJson(filePath));
+            }
+        }
+        return resources;
+    }
+
     /**
      * Main public function
```

Comment:
As we get closer to launch we should have better comments or remove them

Reply:
I added jsdoc comments to this function. 
Also made all other class methods private so that it is obvious that this is the "Main public function" because typescript.